lib/README.md

```diff
@@ -0,0 +1,127 @@
+## Conduit - core library
```

Comment:
Could we move this library into the ocamldoc for `conduit`, so it shows up in the docs.mirage.io and others? Conduit 2 used an index.mld file for this